### PR TITLE
#RI-2450 - add tutorials provider

### PR DESCRIPTION
**Pull request by AmirAllayarovSofteq**

### CODE REVIEW

File: redisinsight/api/config/default.ts [hunk 1]
```diff
@@ -35,7 +39,8 @@ export default {
     globalPrefix: 'api',
     customPluginsUri: '/plugins',
     staticUri: '/static',
-    enablementAreaUri: '/static/workbench',
+    guidesUri: '/static/workbench/guides',
+    tutorialsUri: '/static/workbench/tutorials',
```
Comment (arthosofteq) [inline redisinsight/api/config/default.ts:43]: could you use `/static/tutorials` and `/static/guides` without `workbench`?

File: redisinsight/api/config/default.ts [hunk 2]
```diff
@@ -92,18 +97,25 @@ export default {
...
+    zip: process.env.TUTORIALS_ZIP || 'data.zip',
+    buildInfo: process.env.TUTORIALS_CHECKSUM || buildInfoFileName,
+    devMode: !!process.env.TUTORIALS_DEV_PATH,
+  },
   content: {
     updateUrl: process.env.CONTENT_UPDATE_URL
       || 'https://github.com/RedisInsight/Statics/releases/download/latest',
     zip: process.env.CONTENT_ZIP || 'data.zip',
-    buildInfo: process.env.CONTENT_CHECKSUM || 'build.json',
+    buildInfo: process.env.CONTENT_CHECKSUM || buildInfoFileName,
```
Comment (arthosofteq) [inline redisinsight/api/config/default.ts:118]: If you decided to move `build.json` to a constant why you ignored `data.zip`?

File: redisinsight/api/scripts/default-guides.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  GUIDES_CONFIG.updateUrl,
+  GUIDES_CONFIG.buildInfo,
+)).toString();
+
+const buildInfoPath = join(PATH_CONFIG.defaultGuides, GUIDES_CONFIG.buildInfo);
+
+async function init() {
+  try {
+    // get archive
+    const data = getFile(archiveUrl);
```
Comment (arthosofteq) [inline redisinsight/api/scripts/default-guides.ts:24]: looks like you should `wait` for async function here and everywhere below in this file

File: redisinsight/api/scripts/default-tutorials.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  TUTORIALS_CONFIG.updateUrl,
+  TUTORIALS_CONFIG.buildInfo,
+)).toString();
+
+const buildInfoPath = join(PATH_CONFIG.defaultTutorials, TUTORIALS_CONFIG.buildInfo);
+
+async function init() {
+  try {
+    // get archive
+    const data = getFile(archiveUrl);
```
Comment (arthosofteq) [inline redisinsight/api/scripts/default-tutorials.ts:24]: the same. You missed `await here and below

File: redisinsight/api/src/modules/shared/services/instances-business/auto-discovery.service.ts [hunk 5]
```diff
@@ -1,90 +0,0 @@
-import {
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/shared/services/instances-business/auto-discovery.service.ts:1]: Why did you remove this?

File: redisinsight/api/src/modules/shared/shared.module.ts [hunk 6]
```diff
@@ -43,7 +42,6 @@ const SERVER_CONFIG = config.get('server');
     RedisSentinelBusinessService,
     AutodiscoveryAnalyticsService,
     RedisToolFactory,
-    AutoDiscoveryService,
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/shared/shared.module.ts:46]: Why did you remove this?

**Review by arthosofteq — CHANGES_REQUESTED** [review]

File: redisinsight/api/config/default.ts (hunk 2 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/config/default.ts:118]: Just linter highlight. And i move it to constant string. But yea, i'll update 'data.zip' too

File: redisinsight/api/src/modules/shared/services/instances-business/auto-discovery.service.ts (hunk 5 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/shared/services/instances-business/auto-discovery.service.ts:1]: My fault, i don't know when i did it. Restored it

File: redisinsight/api/src/modules/shared/shared.module.ts (hunk 6 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/shared/shared.module.ts:46]: Same

File: redisinsight/api/package.json [hunk 7]
```diff
@@ -10,9 +10,10 @@
   },
   "scripts": {
     "build:defaults:commands": "ts-node ./scripts/default-commands.ts",
-    "build:defaults:enablement-area": "ts-node ./scripts/default-enablement-area.ts",
+    "build:defaults:guides": "ts-node ./scripts/default-guides.ts",
+    "build:defaults:tutorials": "ts-node ./scripts/default-tutorials.ts",
     "build:defaults:content": "ts-node ./scripts/default-content.ts",
-    "build:defaults": "yarn build:defaults:enablement-area && yarn build:defaults:commands yarn build:defaults:content",
+    "build:defaults": "yarn build:defaults:guides && yarn build:defaults:commands && yarn build:defaults:content",
     "prebuild": "rimraf dist",
```
Comment (arthosofteq) [inline redisinsight/api/package.json:17]: Did you test it locally? I assume that default tutorials will not work because I don't see `build:defaults:tutorials` here

File: redisinsight/api/src/modules/shared/services/instances-business/auto-discovery.service.ts [hunk 8]
```diff
@@ -87,4 +87,4 @@ export class AutoDiscoveryService implements OnModuleInit {
       // ignore error
     }
   }
-}
+}
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/shared/services/instances-business/auto-discovery.service.ts:90]: No newline at the end of file

File: redisinsight/api/src/utils/auto-discovery-helper.spec.ts  [hunk 9]
```diff
@@ -77,4 +77,4 @@ describe('getTCP4Endpoints', () => {
       expect(result).toEqual(test.output);
     });
   });
-});
+});
```
Comment (arthosofteq) [inline redisinsight/api/src/utils/auto-discovery-helper.spec.ts :80]: No new line...

File: redisinsight/api/src/utils/file-helper.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+  const zip = new AdmZip(data);
+  zip.extractAllTo(path, true);
+};
+
+/**
+ * Exract file in folder
+ */
+export const updateFile = async (path: string, fileName: string, data: any): Promise<void> => {
+
```
Comment (arthosofteq) [inline redisinsight/api/src/utils/file-helper.ts:36]: Do you use linter? It should highlite extra line here

**Review by arthosofteq — CHANGES_REQUESTED** [review]

File: redisinsight/api/package.json (hunk 7 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/package.json:17]: I didn't include the tutorials because there are no files in the repository, so the script will fail with an error

**Review by arthosofteq — CHANGES_REQUESTED** [review]

**Review by arthosofteq — APPROVED** [review]
